### ISSUE TITLE
Should setPushToken be called explicitely?

### ISSUE THREAD

**Issue by jfbn** [CleverTap/clevertap-react-native#166]
Upon integrating this library I struggled with getting push notifications to work. Somewhere along the process I stumbled upon[ this part of the JS interface](https://github.com/CleverTap/clevertap-react-native/blob/master/index.js#L111) which describes a method that can be used to manually set the push device token to the Clevertap profile.

I didn't see this mentioned anywhere in the integration and getting started documentation, so I really don't know if I should be manually calling this method or not. How does Clevertap receive the correct push token and associate it with a Clevertap profile, if not via this method? 

I am using FCM for Android. 

In app startup I am invoking the `getToken()` method from [@react-native-firebase/messaging](https://rnfirebase.io/messaging/usage) and then passing the result from that to the `setPushToken` method exposed via the Clevertap interface. Is this incorrect? I have noticed that some user profiles in Clevertap are receiving the `Reachable by` event 2 times at the exact same time, with different tokens. I assume that this is a result of me manually triggering the `setPushToken` event. When I run a debug build where I have removed the manual call to `setPushToken` I don't receive any "Reachable by" events.

TL;DR

What is the correct approach to informing Clevertap of my push device tokens?

**Comment by piyush-kukadiya** [comment]
@jfbn `setPushToken()` is an equivalent of [this native functionality](https://developer.clevertap.com/docs/android#section-custom-android-push-notifications-handling). You can use it if you are custom handling push notification.

**Comment by jfbn** [comment]
> @jfbn `setPushToken()` is an equivalent of [this native functionality](https://developer.clevertap.com/docs/android#section-custom-android-push-notifications-handling). You can use it if you are custom handling push notification.

Thank you for your reply.

So the answer to my last question "What is the correct approach to informing Clevertap of my push device token" is: do nothing, unless I am handling push notifications in a custom manner. Which I don't think I am.

**Comment by piyush-kukadiya** [comment]
Correct

**Comment by smohit02** [comment]
I have a follow up question to this issue.

My requirement is to get the FCM token only from [@react-native-firebase/messaging](https://rnfirebase.io/messaging/usage).

And send this token to Clevertap using CleverTap.setPushToken() without having to write Native Java Code for generating token from the FirebaseMessagingService class.

This is, since I want to receive notifications from the Firebase Console as well as Clevertap Dashboard using the token generated from "messaging().getToken()" react-native-firebase/messaging, as I can control this from my React Native code.

I don't understand why is there a need for the Native FirebaseMessagingService class and all the native code in the MainApplication.java.
If someone can please elaborate this, and what exactly is custom notification handling?
